### DISCUSSION TITLE
Can't understand how to use the QR given after every lesson...

### DISCUSSION THREAD

**Discussion by AbyyyCJL** [Cyfrin/foundry-full-course-cu#1063] [Q&A]
I'm not finding my query in the comments but I'll just post it here. Does anyone knows how to use the QR code? I go to the website in order to solve the challenges but where are they? Struggling to navigate, please help.

**Comment by Niferu** [comment]
Hello @AbyyyCJL 

Once you scan QR code you will be taken to etherscan contract associated with challenge. Your next move should be just checking source code of that contract on etherscan, you will find challenge in contract code :)

**Reply by BLTC-520** [reply]
Sorry if I ask a dumb question, after I find challenge in contract code, what should I do next? (Modifying the code? but I can't do that directly, help : :)

**Reply by Niferu** [reply]
Hi @BLTC-520 

You should solve its challenge function, so basically you should do something to successfully call this function with desired effect.

**Reply by BLTC-520** [reply]
I'm still confused about what I should do after clicking the "Write Contract" Button, I did successfully mint the "Twitter Handle", but the first contract and the third contract I don't know how and where to modify(What I meant here is what should I click to modify the code but not the code problem itself...) 


sorry if it's dumb again...

**Reply by Niferu** [reply]
@BLTC-520 

I would suggest you of finishing course first if you did not yet. And if you did finish course please make sure you understand all concepts and try to redo course by yourself to get more understanding etc. Those challenges are not so easy to solve and if you struggle of solving first 3 it is definitely not time for you to play with them as those are getting much harder later on.

**Reply by BLTC-520** [reply]
Arghhhh.... maybe I didn't explain my problem well 😢 but thanks bro.

**Reply by Niferu** [reply]
@BLTC-520 

You did explain it well but if you have trouble understanding how to call simple function from external contract without using etherscan interface itself you definitely need some more understanding how contracts and functions work. You just need to call challenge contract exact function giving it correct answers/parameters whatever it depends on challenge. There is nothing more.
